ompi/mpi/c/init_thread.c.in

```diff
@@ -39,22 +39,29 @@
 PROTOTYPE ERROR_CLASS init_thread(INT_OUT argc, ARGV argv, INT required,
                                   INT_OUT provided)
 {
-    int err, safe_required = MPI_THREAD_SERIALIZED;
+    int err, err_arg_required = false, safe_required = MPI_THREAD_SERIALIZED;
     char *env;
 
     ompi_hook_base_mpi_init_thread_top(argc, argv, required, provided);
 
     /* Detect an incorrect thread support level, but dont report until we have the minimum
      * infrastructure setup.
      */
-    if( (MPI_THREAD_SINGLE == required) || (MPI_THREAD_SERIALIZED == required) ||
```

Comment:
This is inconsistent with the way this is tested in MPI_INIT, I picked this resolution but we could change MPI_INIT instead to use the strict equalities.